tools/heir-lsp.cpp

```diff
@@ -0,0 +1,51 @@
+#include "include/Dialect/BGV/IR/BGVDialect.h"
+#include "include/Dialect/CGGI/IR/CGGIDialect.h"
+#include "include/Dialect/Comb/IR/CombDialect.h"
+#include "include/Dialect/LWE/IR/LWEDialect.h"
+#include "include/Dialect/Openfhe/IR/OpenfheDialect.h"
+#include "include/Dialect/PolyExt/IR/PolyExtDialect.h"
+#include "include/Dialect/Polynomial/IR/PolynomialDialect.h"
+#include "include/Dialect/Secret/IR/SecretDialect.h"
+#include "include/Dialect/TfheRust/IR/TfheRustDialect.h"
+#include "include/Dialect/TfheRustBool/IR/TfheRustBoolDialect.h"
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"      // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"     // from @llvm-project
+#include "mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/MemRef/IR/MemRef.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/SCF/IR/SCF.h"          // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tosa/IR/TosaOps.h"     // from @llvm-project
+#include "mlir/include/mlir/InitAllDialects.h"             // from @llvm-project
+#include "mlir/include/mlir/Tools/mlir-lsp-server/MlirLspServerMain.h"  // from @llvm-project
+#include "mlir/include/mlir/Tools/mlir-opt/MlirOptMain.h"  // from @llvm-project
+
+using namespace mlir;
+using namespace heir;
+
+int main(int argc, char **argv) {
+  mlir::DialectRegistry registry;
+
+  registry.insert<bgv::BGVDialect>();
+  registry.insert<comb::CombDialect>();
+  registry.insert<lwe::LWEDialect>();
+  registry.insert<cggi::CGGIDialect>();
+  registry.insert<poly_ext::PolyExtDialect>();
+  registry.insert<polynomial::PolynomialDialect>();
+  registry.insert<secret::SecretDialect>();
+  registry.insert<tfhe_rust::TfheRustDialect>();
+  registry.insert<tfhe_rust_bool::TfheRustBoolDialect>();
+  registry.insert<openfhe::OpenfheDialect>();
+
+  // Add expected MLIR dialects to the registry.
+  registry.insert<affine::AffineDialect>();
+  registry.insert<arith::ArithDialect>();
+  registry.insert<func::FuncDialect>();
+  registry.insert<memref::MemRefDialect>();
+  registry.insert<scf::SCFDialect>();
+  registry.insert<tensor::TensorDialect>();
+  registry.insert<tosa::TosaDialect>();
+  registry.insert<LLVM::LLVMDialect>();
```

Comment:
I think you can leave out the upstream dialects, since `registerAllDialects` already handles them https://sourcegraph.com/github.com/llvm/llvm-project@d6c2cbbc6513bd412b34f3bf70e21b5a363b2fd9/-/blob/mlir/include/mlir/InitAllDialects.h?L102
```suggestion
```

Reply:
Updated.